django/utils/autoreload.py

```diff
@@ -276,7 +276,12 @@ def reloader_thread():
 
 def restart_with_reloader():
     while True:
-        args = [sys.executable] + ['-W%s' % o for o in sys.warnoptions] + sys.argv
+        args = [sys.executable] + ['-W%s' % o for o in sys.warnoptions]
+        if sys.argv[0].endswith("django/__main__.py"):
```

Comment:
Out of interest is `endswith` necessary here? To be a bit safer we could compare the `sys.argv[0]` value with the path of `django.__main__`?

Reply:
Thanks for your comment!  It sounds like a bit safer.  I'll try to update comparison add some unit tests.